api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/summary/SummaryHelpersTests.kt

```diff
@@ -0,0 +1,201 @@
+package org.jetbrains.kotlinx.dl.api.core.summary
+
+import org.jetbrains.kotlinx.dl.api.core.shape.TensorShape
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+internal class SummaryHelpersTests {
+    private val sequentialModel = ModelSummary(
+        type = "Sequential",
+        name = "My sequential NN model",
+        layersSummaries = listOf(
+            LayerSummary("input_1", "Input", TensorShape(-1, 28, 28, 1), 0, emptyList()),
+            LayerSummary("conv2d_1", "Conv2D", TensorShape(-1, 28, 28, 32), 832, emptyList()),
+            LayerSummary("maxPool_1", "MaxPool2D", TensorShape(-1, 14, 14, 32), 0, emptyList()),
+            LayerSummary("conv2d_2", "Conv2D", TensorShape(-1, 14, 14, 64), 51264, emptyList()),
+            LayerSummary("maxPool_2", "MaxPool2D", TensorShape(-1, 7, 7, 64), 0, emptyList()),
+            LayerSummary("flatten_1", "Flatten", TensorShape(-1, 3136), 0, emptyList()),
+            LayerSummary("dense_1", "Dense", TensorShape(-1, 512), 1606144, emptyList()),
+            LayerSummary("dense_2", "Dense", TensorShape(-1, 10), 5130, emptyList())
+        ),
+        trainableParamsCount = 1663370,
+        frozenParamsCount = 0
+    )
+
+    private val functionalModel = ModelSummary(
+        type = "Functional",
+        name = "My functional NN model",
+        layersSummaries = listOf(
+            LayerSummary("input_1", "Input", TensorShape(-1, 28, 28, 1), 0, emptyList()),
+            LayerSummary("conv2D_1", "Conv2D", TensorShape(-1, 26, 26, 32), 320, listOf("input_1")),
+            LayerSummary("conv2D_2", "Conv2D", TensorShape(-1, 24, 24, 64), 18496, listOf("conv2D_1")),
+            LayerSummary("maxPool2D", "MaxPool2D", TensorShape(-1, 8, 8, 64), 0, listOf("conv2D_2")),
+            LayerSummary("conv2D_4", "Conv2D", TensorShape(-1, 8, 8, 64), 36928, listOf("maxPool2D")),
+            LayerSummary("conv2D_5", "Conv2D", TensorShape(-1, 8, 8, 64), 36928, listOf("conv2D_4")),
+            LayerSummary("add", "Add", TensorShape(-1, 8, 8, 64), 0, listOf("conv2D_5", "maxPool2D")),
+            LayerSummary("conv2D_6", "Conv2D", TensorShape(-1, 8, 8, 64), 36928, listOf("add")),
+            LayerSummary("conv2D_7", "Conv2D", TensorShape(-1, 8, 8, 64), 36928, listOf("conv2D_6")),
+            LayerSummary("add_1", "Add", TensorShape(-1, 8, 8, 64), 0, listOf("conv2D_7", "add")),
+            LayerSummary("conv2D_8", "Conv2D", TensorShape(-1, 6, 6, 64), 36928, listOf("add_1")),
+            LayerSummary("globalAvgPool2D", "GlobalAvgPool2D", TensorShape(-1, 64), 0, listOf("conv2D_8")),
+            LayerSummary("dense_1", "Dense", TensorShape(-1, 256), 16640, listOf("globalAvgPool2D")),
+            LayerSummary("dense_2", "Dense", TensorShape(-1, 10), 2570, listOf("dense_1"))
+        ),
+        trainableParamsCount = 222666,
+        frozenParamsCount = 911333
```

Comment:
It's a little bit strange to use unrealistic numbers for the frozen parameters, but it's ok for testing the ModelSummary